dotCMS/src/main/java/com/dotcms/ai/api/AIVisionAPI.java

```diff
@@ -0,0 +1,65 @@
+package com.dotcms.ai.api;
+
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.type.DotAssetContentType;
+import com.dotmarketing.portlets.contentlet.business.exporter.ImageFilterExporter;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.json.JSONObject;
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.vavr.Lazy;
+import io.vavr.Tuple2;
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+public interface AIVisionAPI {
+
+    static final String AI_VISION_AUTOTAG_CONTENTTYPES_KEY ="AI_VISION_AUTOTAG_CONTENTTYPES";
+
+    static final String AI_VISION_MODEL = "AI_VISION_MODEL";
+
+    static final String AI_VISION_MAX_TOKENS = "AI_VISION_MAX_TOKENS";
+
+    static final String AI_VISION_PROMPT = "AI_VISION_PROMPT";
+
+    static final String AI_VISION_ALT_FIELD_VAR = "dotAIDescriptionSrc";
+
+    static final String AI_VISION_TAG_FIELD_VAR = "dotAITagSrc";
+
+    static final String AI_VISITON_TAG_AND_ALT_PROMPT_TEMPLATE = "AI_VISITON_TAG_AND_ALT_PROMPT_TEMPLATE";
+
+    static final Lazy<AIVisionAPI> instance = Lazy.of(com.dotcms.ai.api.OpenAIVisionAPIImpl::new);
```

Comment:
I think future forward we have to have such as a factory or so, to be able to select the vendor and model depending on user parameters, but it is ok by now